packages/consensus/source/consensus.ts

```diff
@@ -460,6 +460,8 @@ export class Consensus implements Contracts.Consensus.Service {
 		const block = await registeredProposer.prepareBlock(roundState.proposer.getWalletPublicKey(), this.#round);
 		this.logger.info(`Proposing new block ${this.#height}/${this.#round} with blockId: ${block.data.id}`);
 
+		void this.eventDispatcher.dispatch(Enums.BlockEvent.Forged, block);
```

Comment:
@sebastijankuzner unsure if this is the best place to do this, but in other places where we only have the proposal, trying to access the block via `getData()` throws because `isDataDeserialized` is false.

Reply:
It is a good place.